File: leetcode-everyday/贪心算法/406-根据身高重建队列/406-根据身高重建队列.cpp
class Solution {
public:
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        vector<vector<int>> res;
        sort(people.begin(), people.end(), [](const vector<int>& a, const vector<int>& b) {
            return a[0] > b[0] || (a[0] == b[0] && a[1] < b[1]);
        });
        // [[7, 0], [7, 1], [6,1], [5, 0], [5, 2], [4, 4]]
        for(const vector<int>& person: people) {
            res.insert(res.begin() + person[1], person);
        }
        return res;
    }
};

File: leetcode-everyday/贪心算法/406-根据身高重建队列/406-根据身高重建队列.js
/**
 * @param {number[][]} people
 * @return {number[][]}
 */
 var reconstructQueue = function(people) {
   // 这里js的sort要注意
  people.sort((a, b) => {
      if(a[0] !== b[0]) return b[0] - a[0];
      return a[1] - b[1];
  })
  const res = [];
  for(const person of people) {
      res.splice(person[1], 0, person);
  }
  return res;
};